calcom/cal.com
# fix: Disallow changing username and email in case of Organization email invite

## What does this PR do?

Fixes following issues - Regression from https://github.com/calcom/cal.com/pull/11421/files#diff-e7c1df7ced03badb4ff3248d2f29752d2a9c968747c96b179aadc08e350d7a1eL166
- Member Invite: Invited member was able to change the email and username
- Invite Link: Person signing up using invite link was required to provide username and he could potentially provide a different username than  the one derived from email.

Also, adds more assertions and flows in existing signup tests
- Verifies that username is correctly derived from email when the email is of the org.
- Verifies that username is correctly derived from email when the email is not of the org.
- Verifies that non-org signup uses the provided username.

Implementation Details:
`username` validation and computation has been added at backend as well so that even if username is changed at backend, it would ignore it in case of organization invite.


Reported at https://threads.com/thread/34551320020/34551320023?s=f5bBRyt2WeiKaHf1zCFK57

Member Invite - Now:
![image](https://github.com/calcom/cal.com/assets/1780212/2214cdbd-d407-46c2-b25a-2c7ecd5c78f7)
Member Invite - Before:
![image](https://github.com/calcom/cal.com/assets/1780212/759bf9a4-4202-41ff-b6c7-a14bc24822c4)

Invite Link - Now
![image](https://github.com/calcom/cal.com/assets/1780212/e4ac6515-14d8-4f31-959f-43fe04464224)
Invite Link -Before
![image](https://github.com/calcom/cal.com/assets/1780212/e12f5827-832a-4f67-99db-249772af8ee9)


## Type of change

## How should this be tested?
- Invite an email to an org say member1@gmail.com
- Go to the email received and reach the signup page from the link in the email
- Observe the username and email fields. They should be disabled.

## Mandatory Tasks
